### PR TITLE
Subscribe lecture button fixes #197

### LINKED ISSUE #197: Subscribe button on no access page

### PR DESCRIPTION
**Pull request by henrixapp**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Feature

* **Please check if the PR fulfills these requirements**

- [x] E2E Tests for the changes have been added  via Cypress

* **What is the current behavior?** (You can also link to an open issue here)

#197 

* **What is the new behavior (if this is a feature change)?**

You can try to subscribe on to a lecture on its page.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No.

* **Other information**:

### CODE REVIEW

**Comment by henrixapp** [comment]
http_caching issue must be resolved.

File: app/models/ability.rb [hunk 1]
```diff
@@ -213,7 +213,7 @@ def initialize(user)
 
       can [:show_announcements, :organizational,
            :show_structures, :search_examples, :search, :show_random_quizzes,
-           :display_course], Lecture
+           :display_course, :subscribe_page], Lecture
```
Comment (fosterfarrell9) [inline app/models/ability.rb:216]: This does the trick only for generic users. The corresponding rules should also be added to admins/editors/teachers (their rulesets are some lines above in the other parts of the `if` block).

File: app/controllers/profile_controller.rb [hunk 2]
```diff
@@ -148,6 +150,7 @@ def email_params
 
   def set_lecture
     @lecture = Lecture.find_by_id(lecture_params[:id])
+    p(@lecture)
```
Comment (fosterfarrell9) [inline app/controllers/profile_controller.rb:153]: This can be removed :-)

File: config/locales/en.yml [hunk 3]
```diff
@@ -2163,6 +2163,7 @@ en:
     click_and_paste: >
       If you click this term, it will be copied to the "fulltext" field.
   errors:
+    not_subscribed: You are currently not subscribed to this course.
```
Comment (fosterfarrell9) [inline config/locales/en.yml:2166]: It should be "You are currently not subscribed to this lecture".

File: config/locales/de.yml [hunk 4]
```diff
@@ -2296,6 +2296,7 @@ de:
     click_and_paste: >
       Wenn Du den Begriff anklickst, wird er in das Suchfeld "Volltext" kopiert.
   errors:
+    not_subscribed: Du bist zur Zeit nicht in dieser Vorlesung AbonnentIn.
```
Comment (fosterfarrell9) [inline config/locales/de.yml:2299]: I would suggest some rewording like "Aktuell hast Du diese Veranstaltung nicht abonniert."

**Review by fosterfarrell9 — CHANGES_REQUESTED** [review]

**Review by fosterfarrell9 — CHANGES_REQUESTED** [review]
The availability of this feature for editors and teachers has yet to be added, is does not work for them as of now: see the comment in the last feature. The reason ist that a `can :subscribe_page, Lecture` is missing for them in `ability.rb` (without that, they are redirected to main with the usual error message).

**Review by fosterfarrell9 — APPROVED** [review]

**Pull request merged**